src/webapi/Controllers/AccountController.cs

```diff
@@ -0,0 +1,217 @@
+ */
+
+[Authorize(AuthenticationSchemes = OpenIddictValidationAspNetCoreDefaults.AuthenticationScheme, Roles = Roles.Administrator)]
+[Route(Route)]
+public class AccountController : Controller
+{
+    public const string Route = "accounts";
+
+    private readonly UserManager<ApplicationUser> _userManager;
+    private readonly IIdentityRepository _identityRepository;
+    private readonly IOpenIddictTokenManager _tokenManager;
+
+    public AccountController(UserManager<ApplicationUser> userManager, IIdentityRepository identityRepository, IOpenIddictTokenManager tokenManager)
+    {
+        _userManager = userManager ?? throw new ArgumentNullException(nameof(userManager));
+        _identityRepository = identityRepository ?? throw new ArgumentNullException(nameof(identityRepository));
+        _tokenManager = tokenManager ?? throw new ArgumentNullException(nameof(tokenManager));
+    }
+
+    [HttpGet("{id}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> Get([FromRoute] string id)
+    {
+        if (id is null || id == string.Empty) { return NotFound(); }
+
+        var result = await _userManager.FindByIdAsync(id);
+        if (result is null || result.DeletedAt is not null) { return NotFound(); }
+
+        return Ok(UserAccountResponse.From(result));
+    }
+
+    [HttpGet()]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+    public async Task<IActionResult> GetAll()
+    {
+        var users = await _identityRepository.FindAllUsers();
+
+        return Ok(users.Select(x => UserAccountResponse.From(x)));
+    }
+
+    [HttpPost]
+    [AllowAnonymous]
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> Post([FromBody] Register model)
+    {
+        if (ModelState.IsValid)
+        {
+            var user = await _userManager.FindByNameAsync(model.Email);
+            if (user is not null)
+            {
+                return EmailConflict();
+            }
+
+            user = model.ToApplicationUser();
+            var result = await _userManager.CreateAsync(user, model.Password);
+
+            if (result.Succeeded)
+            {
+                var roleSuccess = await AddToRole(user);
+                if (!roleSuccess)
+                {
+                    return new JsonResult(ModelState);
+                }
+
+                return Created($"/{Route}/{user.Id}", UserAccountResponse.From(user));
+            }
+
+            AddErrors(result);
+        }
+
+        return BadRequest(new { validationError = ModelState });
+
+
+        async Task<bool> AddToRole(ApplicationUser user)
```

Comment:
I am a fan of local functions in C#. There are at least two good reasons to create a function:

1. Using the same code in multiple places.
2. Encapsulate some logic so that you can simply read the function name without having to read the function details.

This is an example of the second type. The main function body is from lines 69 to 94. I like to keep the main function small like this. If you are reading the function and translating into English (or any other spoken language 😉), then you can easily see from line 80 "If the user was created successfully, then add the user to some role and check the success again". It is not necessary, on the first read through, to see the details of what it takes to add the user to the role. When you do want more detail, you can scroll down to line 97 and read this function.

Creating this as a _local_ function emphasizes that its only reason for being is reason 2, not reason 1 (reuseability).